### ISSUE TITLE
compile error in netbuf v0.2.0

### ISSUE THREAD

**Issue by szagi3891** [tailhook/netbuf#1]
I try run "rotor-http" example file.

When compiling get an error :

```
grzegorz@grzegorz:~/Pulpit/rotor-http$ cargo run --example threaded --verbose
    Updating registry `https://github.com/rust-lang/crates.io-index`
    Updating git repository `git://github.com/tailhook/rotor`
       Fresh unicase v1.0.1
       Fresh slab v0.1.2
       Fresh httparse v1.0.0
       Fresh bytes v0.2.11
       Fresh gcc v0.3.19
       Fresh libc v0.1.12
       Fresh traitobject v0.0.1
       Fresh language-tags v0.0.7
       Fresh log v0.3.3
       Fresh mime v0.1.0
       Fresh hpack v0.2.0
       Fresh solicit v0.4.4
       Fresh winapi v0.2.5
       Fresh bitflags v0.3.3
       Fresh matches v0.1.2
       Fresh libc v0.2.2
       Fresh memchr v0.1.7
       Fresh winapi v0.1.23
       Fresh bitflags v0.1.1
   Compiling nix v0.3.9
     Running `rustc /home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/nix-0.3.9/src/lib.rs --crate-name nix --crate-type lib -g -C metadata=1295479b75f240b4 -C extra-filename=-1295479b75f240b4 --out-dir /home/grzegorz/Pulpit/rotor-http/target/debug/deps --emit=dep-info,link -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps --extern bitflags=/home/grzegorz/Pulpit/rotor-http/target/debug/deps/libbitflags-83dc6fc5ecc9dd71.rlib --extern libc=/home/grzegorz/Pulpit/rotor-http/target/debug/deps/liblibc-6bc3b8388045ae15.rlib --cap-lints allow`
       Fresh lazy_static v0.1.15
   Compiling netbuf v0.2.0
     Running `rustc /home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.0/src/lib.rs --crate-name netbuf --crate-type lib -g -C metadata=f13fe0a853c39504 -C extra-filename=-f13fe0a853c39504 --out-dir /home/grzegorz/Pulpit/rotor-http/target/debug/deps --emit=dep-info,link -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps --cap-lints allow`
/home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.0/src/buf.rs:13:34: 13:41 error: attempted left shift with overflow
/home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.0/src/buf.rs:13 pub const MAX_BUF_SIZE: usize = (1 << 32) - 2;
                                                                                                                           ^~~~~~~
error: aborting due to previous error
Could not compile `netbuf`.

Caused by:
  Process didn't exit successfully: `rustc /home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.0/src/lib.rs --crate-name netbuf --crate-type lib -g -C metadata=f13fe0a853c39504 -C extra-filename=-f13fe0a853c39504 --out-dir /home/grzegorz/Pulpit/rotor-http/target/debug/deps --emit=dep-info,link -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps -L dependency=/home/grzegorz/Pulpit/rotor-http/target/debug/deps --cap-lints allow` (exit code: 101)
```

**Comment by tailhook** [comment]
Should be fixed now. Is it rust nightly? I don't get the error on stable.

**Comment by szagi3891** [comment]
thanks :)

**Comment by szagi3891** [comment]
I still get error :

```
   Compiling netbuf v0.2.2
/home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.2/src/buf.rs:13:34: 13:46 error: attempted left shift with overflow
/home/grzegorz/.cargo/registry/src/github.com-121aea75f9ef2ce2/netbuf-0.2.2/src/buf.rs:13 pub const MAX_BUF_SIZE: usize = (1usize << 32) - 2;
                                                                                                                           ^~~~~~~~~~~~
error: aborting due to previous error
```

**Comment by tailhook** [comment]
Hm, I just realized that you're probably building 32bit version, right?

**Comment by tailhook** [comment]
Could you try netbuf exactly 0.2.1? I think that version should have the issue fixed, I just got the original issue wrong.

**Comment by szagi3891** [comment]
Yes, i have 32bit version system.
How to force this version 0.2.1 ?

**Comment by tailhook** [comment]
@szagi3891 just checkout the version `git checkout v0.2.1` and try to build. If it builds okay, ill backport the fix to the latest version.

**Comment by szagi3891** [comment]
Ok.
It seems that the version 0.2.1 is correct.

```
grzegorz@grzegorz:~/Pulpit/netbuf$ git checkout def8c62f9670f385f70afbc6ce27ef7a3978f0e6
HEAD is now at def8c62... Version bumped to v0.2.1
grzegorz@grzegorz:~/Pulpit/netbuf$ cargo build
   Compiling netbuf v0.2.1 (file:///home/grzegorz/Pulpit/netbuf)
grzegorz@grzegorz:~/Pulpit/netbuf$ 
```

**Comment by tailhook** [comment]
Okay, published v0.2.3

**Comment by szagi3891** [comment]
It works great.
